evcache-core/src/main/java/com/netflix/evcache/EVCacheImpl.java

```diff
@@ -977,7 +969,30 @@ private EVCacheItemMetaData getEVCacheItemMetaData(EVCacheClient client, EVCache
         if (client == null) return null;
         final Transcoder<T> transcoder = (tc == null) ? ((_transcoder == null) ? (Transcoder<T>) client.getTranscoder() : (Transcoder<T>) _transcoder) : tc;
         try {
-            return client.metaGet(evcKey.getDerivedKey(client.isDuetClient()), transcoder, throwException, hasZF);
+            String hashKey = evcKey.getHashKey(client.isDuetClient(), client.getHashingAlgorithm());
+            String canonicalKey = evcKey.getCanonicalKey(client.isDuetClient());
+            if (hashKey != null) {
+                final EVCacheItem<Object> obj = client.metaGet(hashKey, evcacheValueTranscoder, throwException, hasZF);
+                if (obj.getData() instanceof EVCacheValue) {
+                    final EVCacheValue val = (EVCacheValue) obj.getData();
+                    if (null == val) {
+                        return null;
+                    }
+                    if (!(val.getKey().equals(canonicalKey))) {
+                        incrementFailure(EVCacheMetricsFactory.KEY_HASH_COLLISION, Call.GET.name(), EVCacheMetricsFactory.META_GET_OPERATION);
```

Comment:
@smadappa As I pulled this code from EVCacheClient to Impl, I added the operationType, in addition to the earlier existing operation. jfyi.

Reply:
@sumanth-pasupuleti Should we change this to `Call.META_GET.name()`?

Reply:
That makes a lot more sense! Changing it to Call.META_GET.name()